sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/options/BlobBatchSetBlobAccessTierOptions.java

```diff
@@ -89,7 +89,8 @@ public String getBlobName() {
      * @return Identifier of the blob to set its access tier.
      */
     public String getBlobIdentifier() {
-        String basePath = blobUrlParts.getBlobContainerName() + "/" + blobUrlParts.getBlobName();
+        String basePath = Utility.urlEncode(blobUrlParts.getBlobContainerName()) + "/"
+            + Utility.urlEncode(blobUrlParts.getBlobName());
```

Comment:
How does this method get used in the SDK or how would customers use this to set access tiers with the SDK? I'm a little hesitant to introduce encodings especially on the blob name if we had not been doing it before, but I'm not sure what a blob identifier is supposed to represent conceptually and how customers use this.